pandas/core/indexes/datetimes.py

```diff
@@ -864,11 +864,18 @@ def _add_datelike(self, other):
     def _sub_datelike(self, other):
         # subtract a datetime from myself, yielding a TimedeltaIndex
         from pandas import TimedeltaIndex
+
+        if isinstance(other, np.ndarray):
```

Comment:
why don't you do this in 873, and just always wrap it

Reply:
Sounds good.